### PR TITLE
Checkpoint bug fix

### LINKED ISSUE #1608: multiband_melgan Vocoder Fails on Step 10000 With KeyError: 'avg_loss_0'
### Describe the bug

[multiband_melgan](https://github.com/coqui-ai/TTS/blob/dev/recipes/ljspeech/multiband_melgan/train_multiband_melgan.py) vocoder trained on [en](https://data.solak.de/data/Training/stt_tts/en_US.tgz), specifically, `"by_book/male/elliot_miller/pirates_of_ersatz/"` (tried another language, failed with the same error), fails on step 10000 with KeyError: 'avg_loss_0'.

### To Reproduce

Run train_multiband_melgan.py on en_US dataset.

### Expected behavior

No Errors.

### Logs

_No response_

### Environment

```shell
{
    "CUDA": {
        "GPU": [
            "Tesla T4",
            "Tesla T4",
            "Tesla T4",
            "Tesla T4"
        ],
        "available": true,
        "version": "10.2"
    },
    "Packages": {
        "PyTorch_debug": false,
        "PyTorch_version": "1.11.0+cu102",
        "TTS": "0.6.2",
        "numpy": "1.21.6"
    },
    "System": {
        "OS": "Linux",
        "architecture": [
            "64bit",
            "ELF"
        ],
        "processor": "x86_64",
        "python": "3.7.13",
        "version": "#1 SMP Tue Apr 26 20:14:22 UTC 2022"
    }
}
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by manmay-nakhashi**
#1608

### CODE REVIEW

**Comment by erogol** [comment]
I am not sure how ```_log``` fixes the checkpoint problem?

**Comment by manmay-nakhashi** [comment]
@erogol original error is in trainer.py i have also made a pl over there , after resolving that outputs[0] is coming as None in _log so this is the fix over here.

File: TTS/vocoder/models/gan.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def _log(name: str, ap: AudioProcessor, batch: Dict, outputs: Dict) -> Tuple[Dic
         Returns:
             Tuple[Dict, Dict]: log figures and audio samples.
         """
-        y_hat = outputs[0]["model_outputs"]
+        y_hat = outputs[0]["model_outputs"] if outputs[0] is not None else outputs[1]["model_outputs"]
```
Comment (erogol) [inline TTS/vocoder/models/gan.py:201]: I guess the condition we need to check `self.train_disc` is right? The only case there are no outputs[0] is when `self.train_disc` is False.

**Review by erogol — CHANGES_REQUESTED** [review]

File: TTS/vocoder/models/gan.py (hunk 1 above)
Comment (manmay-nakhashi) [reply TTS/vocoder/models/gan.py]: updated

**Comment by erogol** [comment]
Thanks !!

**Pull request merged**